Add Logs data stream for collecting Azure Functions

devamanv opened the pull request:
<!-- Type of change
Please label this PR with one of the following labels, depending on the scope of your change:
- Bug
- Enhancement
- Breaking change
- Deprecation
-->

## What does this PR do?
The PR contains changes to add a new datastream `functionapplogs` that contains all the field mappings, ingest pipelines, documentations needed to ingest the Azure Functions logs into Elasticsearch. 

Note: Dashboards assets to be added in a separate PR. 

<!-- Mandatory
Explain here the changes you made on the PR.
-->

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

- Clone the integrations repo
- Install elastic-package locally
- Spin up an elastic stack using `elastic-package`
- Run `elastic-package test` from the integrations/packages/azure_functions directory

## Related issues
- Implements logs datastream for #6086 

## Screenshots
![image](https://github.com/elastic/integrations/assets/38116245/4d677489-0d44-4120-868b-d3acc22d39ae) 
<img width="1909" alt="image" src="https://github.com/elastic/integrations/assets/38116245/22929a83-ee06-4d9c-a679-6a2de2050edb">

elasticmachine commented:
## :broken_heart: Build Failed


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-6417/detail/PR-6417/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-6417/detail/PR-6417/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-6417/detail/PR-6417/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-6417/detail/PR-6417/1//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://integrations_6417.docs-preview.app.elstc.co/diff)  [![preview](https://img.shields.io/badge/elastic-observability-blue)](https://ci-stats.elastic.co/app/apm/services/fleet-ci/transactions/view?rangeFrom=2023-06-01T03:55:30.736Z&rangeTo=2023-06-01T04:15:30.736Z&transactionName=BUILD+Ingest-manager%2Fintegrations%2FPR-%7Bnumber%7D&transactionType=job&latencyAggregationType=avg&traceId=4461a55bcc195e26ce9488ec795df96a&transactionId=066d253aa081ae34)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats




* Start Time: 2023-06-01T04:05:30.736+0000


* Duration: 5 min 15 sec





</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY -->














### Steps errors [![1](https://img.shields.io/badge/1%20-red)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-6417/detail/PR-6417/1//pipeline)
  <details><summary>Expand to view the steps failures</summary>
  <p>


  
  
  
  
##### `Checks and builds Go sources`
<ul>
<li>Took 0 min 24 sec . View more details <a href="https://fleet-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Ingest-manager/pipelines/integrations/pipelines/PR-6417/runs/1/steps/140/log/?start=0">here</a></li>
<li>Description: <code>mage -debug check</code></l1>
</ul>
  
  </p>
  </details>



## :robot: GitHub comments

<details><summary>Expand to view the GitHub comments</summary>
<p>

To re-run your PR in the CI, just comment with:



- `/test` : Re-trigger the build.




</p>
</details>

<!--COMMENT_GENERATED_WITH_ID_comment.id-->

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/114)
* Commit: 9492f140c153e925eaa62f342f59f3d291e0752f

### Failed CI Steps
* [:pipeline::arrow_up: Upload Pipeline: .buildkite/pipeline.yml](https://buildkite.com/elastic/integrations/builds/114#0188bc34-d635-4b22-9389-f38056446fc0)

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/114","number":114,"commit":"9492f140c153e925eaa62f342f59f3d291e0752f"}],"number":114}
buildkite-pr-comment-->

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/115)
* Commit: 2513ce253aa25b2e0d21857f9d0f1f2e26657280

### Failed CI Steps
* [:pipeline::arrow_up: Upload Pipeline: .buildkite/pipeline.yml](https://buildkite.com/elastic/integrations/builds/115#0188bc44-0ad3-42c0-a1db-00a87b193e7e)

### History
* :broken_heart: [Build #114](https://buildkite.com/elastic/integrations/builds/114) failed 9492f140c153e925eaa62f342f59f3d291e0752f

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/115","number":115,"commit":"2513ce253aa25b2e0d21857f9d0f1f2e26657280"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/114","number":114,"commit":"9492f140c153e925eaa62f342f59f3d291e0752f"}],"number":115}
buildkite-pr-comment-->

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/137)
* Commit: 43049e531c6b4af9ac55205376bd1e4f1631064f

### Failed CI Steps
* [:pipeline::arrow_up: Upload Pipeline: .buildkite/pipeline.yml](https://buildkite.com/elastic/integrations/builds/137#0188bee1-053f-40b3-81a0-9a0960eb1615)

### History
* :broken_heart: [Build #115](https://buildkite.com/elastic/integrations/builds/115) failed 2513ce253aa25b2e0d21857f9d0f1f2e26657280
* :broken_heart: [Build #114](https://buildkite.com/elastic/integrations/builds/114) failed 9492f140c153e925eaa62f342f59f3d291e0752f

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/137","number":137,"commit":"43049e531c6b4af9ac55205376bd1e4f1631064f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/115","number":115,"commit":"2513ce253aa25b2e0d21857f9d0f1f2e26657280"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/114","number":114,"commit":"9492f140c153e925eaa62f342f59f3d291e0752f"}],"number":137}
buildkite-pr-comment-->

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/139)
* Commit: 148085d0ea3e0ff212e24221d64793a9e8bba99e

### Failed CI Steps
* [:pipeline::arrow_up: Upload Pipeline: .buildkite/pipeline.yml](https://buildkite.com/elastic/integrations/builds/139#0188beee-8fdb-4e10-9b8c-6acb22daf75b)

### History
* :broken_heart: [Build #137](https://buildkite.com/elastic/integrations/builds/137) failed 43049e531c6b4af9ac55205376bd1e4f1631064f
* :broken_heart: [Build #115](https://buildkite.com/elastic/integrations/builds/115) failed 2513ce253aa25b2e0d21857f9d0f1f2e26657280
* :broken_heart: [Build #114](https://buildkite.com/elastic/integrations/builds/114) failed 9492f140c153e925eaa62f342f59f3d291e0752f

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/139","number":139,"commit":"148085d0ea3e0ff212e24221d64793a9e8bba99e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/137","number":137,"commit":"43049e531c6b4af9ac55205376bd1e4f1631064f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/115","number":115,"commit":"2513ce253aa25b2e0d21857f9d0f1f2e26657280"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/114","number":114,"commit":"9492f140c153e925eaa62f342f59f3d291e0752f"}],"number":139}
buildkite-pr-comment-->

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/140)
* Commit: 3b03d249cf48dde8e8fed71ca49aee2beef2b181

### Failed CI Steps
* [:pipeline::arrow_up: Upload Pipeline: .buildkite/pipeline.yml](https://buildkite.com/elastic/integrations/builds/140#0188bef7-9c4f-4205-9ef5-e5aa91436b05)

### History
* :broken_heart: [Build #139](https://buildkite.com/elastic/integrations/builds/139) failed 148085d0ea3e0ff212e24221d64793a9e8bba99e
* :broken_heart: [Build #137](https://buildkite.com/elastic/integrations/builds/137) failed 43049e531c6b4af9ac55205376bd1e4f1631064f
* :broken_heart: [Build #115](https://buildkite.com/elastic/integrations/builds/115) failed 2513ce253aa25b2e0d21857f9d0f1f2e26657280
* :broken_heart: [Build #114](https://buildkite.com/elastic/integrations/builds/114) failed 9492f140c153e925eaa62f342f59f3d291e0752f

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/140","number":140,"commit":"3b03d249cf48dde8e8fed71ca49aee2beef2b181"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/139","number":139,"commit":"148085d0ea3e0ff212e24221d64793a9e8bba99e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/137","number":137,"commit":"43049e531c6b4af9ac55205376bd1e4f1631064f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/115","number":115,"commit":"2513ce253aa25b2e0d21857f9d0f1f2e26657280"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/114","number":114,"commit":"9492f140c153e925eaa62f342f59f3d291e0752f"}],"number":140}
buildkite-pr-comment-->

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/148)
* Commit: 0ff493b19dd636f883abc9c7b37f29b97cd8958e

### Failed CI Steps
* [:pipeline::arrow_up: Upload Pipeline: .buildkite/pipeline.yml](https://buildkite.com/elastic/integrations/builds/148#0188bf48-f27d-4496-a1d4-cae0c8ddcad4)

### History
* :broken_heart: [Build #143](https://buildkite.com/elastic/integrations/builds/143) failed 3c1b8328561dc7fedf8fe0eb75e3675c7c1e22a8
* :broken_heart: [Build #140](https://buildkite.com/elastic/integrations/builds/140) failed 3b03d249cf48dde8e8fed71ca49aee2beef2b181
* :broken_heart: [Build #139](https://buildkite.com/elastic/integrations/builds/139) failed 148085d0ea3e0ff212e24221d64793a9e8bba99e
* :broken_heart: [Build #137](https://buildkite.com/elastic/integrations/builds/137) failed 43049e531c6b4af9ac55205376bd1e4f1631064f
* :broken_heart: [Build #115](https://buildkite.com/elastic/integrations/builds/115) failed 2513ce253aa25b2e0d21857f9d0f1f2e26657280
* :broken_heart: [Build #114](https://buildkite.com/elastic/integrations/builds/114) failed 9492f140c153e925eaa62f342f59f3d291e0752f

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/148","number":148,"commit":"0ff493b19dd636f883abc9c7b37f29b97cd8958e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/143","number":143,"commit":"3c1b8328561dc7fedf8fe0eb75e3675c7c1e22a8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/140","number":140,"commit":"3b03d249cf48dde8e8fed71ca49aee2beef2b181"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/139","number":139,"commit":"148085d0ea3e0ff212e24221d64793a9e8bba99e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/137","number":137,"commit":"43049e531c6b4af9ac55205376bd1e4f1631064f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/115","number":115,"commit":"2513ce253aa25b2e0d21857f9d0f1f2e26657280"}],"number":148}
buildkite-pr-comment-->

zmoog commented on packages/azure_functions/manifest.yml:
```diff
@@ -0,0 +1,86 @@
+  - azure
+  - cloud
+conditions:
+  kibana.version: "^8.6.1"
```
```suggestion
  kibana.version: "^8.8.1"
```

We need at least 8.8.1 to use the JSON santitization feature added in this version.

devamanv replied:
Good point, will change that.

zmoog reviewed (changes requested):
This is good, I just have a few minor change requests.

devamanv commented on packages/azure_functions/LICENSE.txt:
When we use `elastic-package` to create an integration package, the CLI goes through a few prompts, one of them is a license(as seen below)
<img width="1354" alt="image" src="https://github.com/elastic/integrations/assets/38116245/0fbfe281-1fa4-4a2f-9f45-c28f081c61cd">

zmoog commented on packages/azure_functions/data_stream/functionapplogs/agent/stream/azure-eventhub.yml.hbs:
```diff
@@ -0,0 +1,41 @@
+processors:
+{{processors}}
+{{/if}}
+sanitize_options:
```
Let me know if you need clarification; I will be more than happy to help.

zmoog commented on packages/azure_functions/docs/README.md:
```diff
@@ -0,0 +1,159 @@
+
+| Log Category                 | Description                                                                                                                          |
+|:----------------------------:|:------------------------------------------------------------------------------------------------------------------------------------:|
+| functionapplogs | Function app logs.        |
```
Resolving this as this file will be re-generated when we update `packages/azure_functions/_dev/build/docs/README.md`

devamanv commented on packages/azure_functions/data_stream/functionapplogs/elasticsearch/ingest_pipeline/default.yml:
```diff
@@ -0,0 +1,186 @@
+      field: json.properties.eventID
+      target_field: event.code
+      ignore_missing: true      
+  - set:
+      field: json.event.alert.action
+      value: denied
+      if: ctx.azure?.firewall?.action == 'Deny'
+  - set:
+      field: json.event.alert.action
+      value: allowed
+      if: ctx.azure?.firewall?.action == 'Allow'
+  - append:
+      field: event.type
+      value: "allowed"
+      if: 'ctx.event?.action != null && ["Allowed","Matched","Detected"].contains(ctx.event.action)'
+  - append:
+      field: event.type
+      value: "denied"
+      if: 'ctx.event?.action != null && ["Blocked"].contains(ctx.event.action)'
```
It would seem, yes. Removed the fields, they are not for Azure functions.

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/152)
* Commit: 9eb38d23c07ee70819664301703e950439d78745

### Failed CI Steps
* [:pipeline::arrow_up: Upload Pipeline: .buildkite/pipeline.yml](https://buildkite.com/elastic/integrations/builds/152#0188bfc9-4115-4cdd-852d-64d68d823d49)

### History
* :broken_heart: [Build #148](https://buildkite.com/elastic/integrations/builds/148) failed 0ff493b19dd636f883abc9c7b37f29b97cd8958e
* :broken_heart: [Build #143](https://buildkite.com/elastic/integrations/builds/143) failed 3c1b8328561dc7fedf8fe0eb75e3675c7c1e22a8
* :broken_heart: [Build #140](https://buildkite.com/elastic/integrations/builds/140) failed 3b03d249cf48dde8e8fed71ca49aee2beef2b181
* :broken_heart: [Build #139](https://buildkite.com/elastic/integrations/builds/139) failed 148085d0ea3e0ff212e24221d64793a9e8bba99e
* :broken_heart: [Build #137](https://buildkite.com/elastic/integrations/builds/137) failed 43049e531c6b4af9ac55205376bd1e4f1631064f
* :broken_heart: [Build #115](https://buildkite.com/elastic/integrations/builds/115) failed 2513ce253aa25b2e0d21857f9d0f1f2e26657280

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/152","number":152,"commit":"9eb38d23c07ee70819664301703e950439d78745"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/148","number":148,"commit":"0ff493b19dd636f883abc9c7b37f29b97cd8958e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/143","number":143,"commit":"3c1b8328561dc7fedf8fe0eb75e3675c7c1e22a8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/140","number":140,"commit":"3b03d249cf48dde8e8fed71ca49aee2beef2b181"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/139","number":139,"commit":"148085d0ea3e0ff212e24221d64793a9e8bba99e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/137","number":137,"commit":"43049e531c6b4af9ac55205376bd1e4f1631064f"}],"number":152}
buildkite-pr-comment-->

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/154)
* Commit: b49595c647e287d687490f6e0aeaaf996146b5ff

### Failed CI Steps
* [:pipeline::arrow_up: Upload Pipeline: .buildkite/pipeline.yml](https://buildkite.com/elastic/integrations/builds/154#0188bff7-abad-43f5-8100-5241a50b8585)

### History
* :broken_heart: [Build #153](https://buildkite.com/elastic/integrations/builds/153) failed 891caf7479ffe33436aaef9137ce6b43702b08d9
* :broken_heart: [Build #152](https://buildkite.com/elastic/integrations/builds/152) failed 9eb38d23c07ee70819664301703e950439d78745
* :broken_heart: [Build #148](https://buildkite.com/elastic/integrations/builds/148) failed 0ff493b19dd636f883abc9c7b37f29b97cd8958e
* :broken_heart: [Build #143](https://buildkite.com/elastic/integrations/builds/143) failed 3c1b8328561dc7fedf8fe0eb75e3675c7c1e22a8
* :broken_heart: [Build #140](https://buildkite.com/elastic/integrations/builds/140) failed 3b03d249cf48dde8e8fed71ca49aee2beef2b181
* :broken_heart: [Build #139](https://buildkite.com/elastic/integrations/builds/139) failed 148085d0ea3e0ff212e24221d64793a9e8bba99e

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/154","number":154,"commit":"b49595c647e287d687490f6e0aeaaf996146b5ff"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/153","number":153,"commit":"891caf7479ffe33436aaef9137ce6b43702b08d9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/152","number":152,"commit":"9eb38d23c07ee70819664301703e950439d78745"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/148","number":148,"commit":"0ff493b19dd636f883abc9c7b37f29b97cd8958e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/143","number":143,"commit":"3c1b8328561dc7fedf8fe0eb75e3675c7c1e22a8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/140","number":140,"commit":"3b03d249cf48dde8e8fed71ca49aee2beef2b181"}],"number":154}
buildkite-pr-comment-->

elasticmachine commented:
### :globe_with_meridians: Coverage report

Name | Metrics % (`covered/total`) | Diff
 --- | --- | ---
Packages | 100.0% (`1/1`) | :green_heart: 
Files | 100.0% (`2/2`) | :green_heart: 
Classes | 100.0% (`2/2`) | :green_heart: 
Methods | 85.714% (`12/14`) | :grey_exclamation: 
Lines | 89.222% (`149/167`) | :grey_exclamation: 
Conditionals | 100.0% (`0/0`) | :green_heart: 

<!--COMMENT_GENERATED_WITH_ID_coverage-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/155)
* Commit: 4d2256407139584a0723cd8aafde5575e607b5e9

### History
* :broken_heart: [Build #154](https://buildkite.com/elastic/integrations/builds/154) failed b49595c647e287d687490f6e0aeaaf996146b5ff
* :broken_heart: [Build #153](https://buildkite.com/elastic/integrations/builds/153) failed 891caf7479ffe33436aaef9137ce6b43702b08d9
* :broken_heart: [Build #152](https://buildkite.com/elastic/integrations/builds/152) failed 9eb38d23c07ee70819664301703e950439d78745
* :broken_heart: [Build #148](https://buildkite.com/elastic/integrations/builds/148) failed 0ff493b19dd636f883abc9c7b37f29b97cd8958e
* :broken_heart: [Build #143](https://buildkite.com/elastic/integrations/builds/143) failed 3c1b8328561dc7fedf8fe0eb75e3675c7c1e22a8
* :broken_heart: [Build #140](https://buildkite.com/elastic/integrations/builds/140) failed 3b03d249cf48dde8e8fed71ca49aee2beef2b181

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/155","number":155,"commit":"4d2256407139584a0723cd8aafde5575e607b5e9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/154","number":154,"commit":"b49595c647e287d687490f6e0aeaaf996146b5ff"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/153","number":153,"commit":"891caf7479ffe33436aaef9137ce6b43702b08d9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/152","number":152,"commit":"9eb38d23c07ee70819664301703e950439d78745"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/148","number":148,"commit":"0ff493b19dd636f883abc9c7b37f29b97cd8958e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/143","number":143,"commit":"3c1b8328561dc7fedf8fe0eb75e3675c7c1e22a8"}],"number":155}
buildkite-pr-comment-->

gpop63 commented on packages/azure_functions/data_stream/functionapplogs/fields/fields.yml:
```diff
@@ -0,0 +1,60 @@
+- name: azure
```
Shouldn't this be something like `azure.functionapplogs`?

zmoog commented on packages/azure_functions/_dev/build/docs/README.md:
````diff
@@ -0,0 +1,159 @@
+            },
+            "tags": [
+                "preserve_original_event"
+            ]
+        }
+```
+
+**Exported fields**
+
+| Field | Description | Type |
+|---|---|---|
+| @timestamp | Event timestamp. | date |
+| azure.function.app_name | The Function application name. | keyword |
+| azure.function.category | Category | keyword |
+| azure.function.event_name | The event name. | keyword |
+| azure.function.function_invocation_id | The invocation ID that logged the message | keyword |
+| azure.function.function_name_ | The name of the Azure function that logged the message | keyword |
+| azure.function.host_instance_id | The host instance ID. | keyword |
+| azure.function.host_version | The functions host version | keyword |
+| azure.function.level | The log level. Valid values are Trace, Debug, Information, Warning, Error, or Critical. | keyword |
+| azure.function.level_id | The integer value of the log level. Valid values are 0 (Trace), 1 (Debug), 2 (Information), 3 (Warning), 4 (Error), or 5 (Critical). | long |
+| azure.function.message | The log message specific to the function | keyword |
+| azure.function.process_id | The process ID | long |
+| azure.function.role_instance | The role instance ID | keyword |
+| azure.resource.group | Resource group | keyword |
+| azure.resource.id | Resource ID | keyword |
+| azure.resource.name | Name | keyword |
+| azure.resource.namespace | Resource type/namespace | keyword |
+| azure.resource.provider | Resource type/namespace | keyword |
+| azure.subscription_id | Azure subscription ID | keyword |
+| azure.tenant_id | tenant ID | keyword |
+| cloud.account.id | The cloud account or organization id used to identify different entities in a multi-tenant environment. Examples: AWS account id, Google Cloud ORG Id, or other unique identifier. | keyword |
+| cloud.provider | Name of the cloud provider. Example values are aws, azure, gcp, or digitalocean. | keyword |
+| data_stream.dataset | Data stream dataset. | constant_keyword |
+| data_stream.type | Data stream type. | constant_keyword |
+| ecs.version | ECS version this event conforms to. `ecs.version` is a required field and must exist in all events. When querying across multiple indices -- which may conform to slightly different ECS versions -- this field lets integrations adjust to the schema version of the events. | keyword |
+| message | For log events the message field contains the log message, optimized for viewing in a log viewer. For structured logs without an original message field, other fields can be concatenated to form a human-readable summary of the event. If multiple messages exist, they can be combined into one message. | match_only_text |
+| observer.product | The product name of the observer. | keyword |
+| observer.type | The type of the observer the data is coming from. There is no predefined list of observer types. Some examples are `forwarder`, `firewall`, `ids`, `ips`, `proxy`, `poller`, `sensor`, `APM server`. | keyword |
+| observer.vendor | Vendor name of the observer. | keyword |
````
Instead of inlining the fields reference and the sample event inline, we should:

1. Replace this text with template variables; see https://github.com/elastic/integrations/blob/9495cd71789829009efca5b220f4e5204606d8ef/packages/azure_frontdoor/_dev/build/docs/README.md?plain=1#L55-L61 as an example.

2. Add the file `packages/azure_functions/data_stream/functionapplogs/sample_event.json` with the sample event.

The `elastic-package` tool will take care of rendering the source docs into `packages/azure_functions/docs/README.md`

devamanv replied:
Pushed the change, thanks for the suggestion.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/171)
* Commit: 9568fab175c6d73544e21c4c2cd2aefc3fb6c38e

### History
* :green_heart: [Build #155](https://buildkite.com/elastic/integrations/builds/155) succeeded 4d2256407139584a0723cd8aafde5575e607b5e9
* :broken_heart: [Build #154](https://buildkite.com/elastic/integrations/builds/154) failed b49595c647e287d687490f6e0aeaaf996146b5ff
* :broken_heart: [Build #153](https://buildkite.com/elastic/integrations/builds/153) failed 891caf7479ffe33436aaef9137ce6b43702b08d9
* :broken_heart: [Build #152](https://buildkite.com/elastic/integrations/builds/152) failed 9eb38d23c07ee70819664301703e950439d78745
* :broken_heart: [Build #148](https://buildkite.com/elastic/integrations/builds/148) failed 0ff493b19dd636f883abc9c7b37f29b97cd8958e
* :broken_heart: [Build #143](https://buildkite.com/elastic/integrations/builds/143) failed 3c1b8328561dc7fedf8fe0eb75e3675c7c1e22a8

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/171","number":171,"commit":"9568fab175c6d73544e21c4c2cd2aefc3fb6c38e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/155","number":155,"commit":"4d2256407139584a0723cd8aafde5575e607b5e9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/154","number":154,"commit":"b49595c647e287d687490f6e0aeaaf996146b5ff"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/153","number":153,"commit":"891caf7479ffe33436aaef9137ce6b43702b08d9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/152","number":152,"commit":"9eb38d23c07ee70819664301703e950439d78745"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/148","number":148,"commit":"0ff493b19dd636f883abc9c7b37f29b97cd8958e"}],"number":171}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/178)
* Commit: acf02dffe93c23ec5e83ae622095c354ff1a1370

### History
* :green_heart: [Build #171](https://buildkite.com/elastic/integrations/builds/171) succeeded 9568fab175c6d73544e21c4c2cd2aefc3fb6c38e
* :green_heart: [Build #155](https://buildkite.com/elastic/integrations/builds/155) succeeded 4d2256407139584a0723cd8aafde5575e607b5e9
* :broken_heart: [Build #154](https://buildkite.com/elastic/integrations/builds/154) failed b49595c647e287d687490f6e0aeaaf996146b5ff
* :broken_heart: [Build #153](https://buildkite.com/elastic/integrations/builds/153) failed 891caf7479ffe33436aaef9137ce6b43702b08d9
* :broken_heart: [Build #152](https://buildkite.com/elastic/integrations/builds/152) failed 9eb38d23c07ee70819664301703e950439d78745
* :broken_heart: [Build #148](https://buildkite.com/elastic/integrations/builds/148) failed 0ff493b19dd636f883abc9c7b37f29b97cd8958e

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/178","number":178,"commit":"acf02dffe93c23ec5e83ae622095c354ff1a1370"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/171","number":171,"commit":"9568fab175c6d73544e21c4c2cd2aefc3fb6c38e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/155","number":155,"commit":"4d2256407139584a0723cd8aafde5575e607b5e9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/154","number":154,"commit":"b49595c647e287d687490f6e0aeaaf996146b5ff"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/153","number":153,"commit":"891caf7479ffe33436aaef9137ce6b43702b08d9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/152","number":152,"commit":"9eb38d23c07ee70819664301703e950439d78745"}],"number":178}
buildkite-pr-comment-->

zmoog reviewed (approved):
LGTM.

zmoog commented on packages/azure_functions/data_stream/functionapplogs/manifest.yml:
```diff
@@ -0,0 +1,59 @@
+          Processors are used to reduce the number of fields in the exported event or to enhance the event with metadata. This executes in the agent before the logs are parsed. See [Processors](https://www.elastic.co/guide/en/beats/filebeat/current/filtering-and-enhancing-data.html) for details.
+      - name: sanitize_newlines
+        title: Sanitize New Lines
+        description: Sanitizes New Lines
```
Align the description with #6357 

```suggestion
        description: Removes new lines in logs to ensure proper formatting of JSON data and avoid parsing issues during processing.
```

shmsr commented on packages/azure_functions/data_stream/functionapplogs/elasticsearch/ingest_pipeline/default.yml:
```diff
@@ -0,0 +1,170 @@
+      on_failure:
+        - set:
+            field: error.message
+            value: 'Received invalid json from the Azure Function service. Unable to parse the source log message'
```
```suggestion
            value: 'Received invalid JSON from the Azure Function service. Unable to parse the source log message'
```

shmsr commented on packages/azure_functions/data_stream/functionapplogs/manifest.yml:
```diff
@@ -0,0 +1,59 @@
+        default: false
+      - name: sanitize_singlequotes
+        title: Sanitize Single Quotes
+        description: This toggle is used to sanitize the event data by replacing single quotes with double quotes, which otherwise may result in pipeline failures. This option is set to True by default for Azure Functions integration due to a tendency of Azure Cloud to send json fields with strings in single quotes.
```
```suggestion
        description: This toggle is used to sanitize the event data by replacing single quotes with double quotes, which otherwise may result in pipeline failures. This option is set to True by default for Azure Functions integration due to a tendency of Azure Cloud to send JSON fields with strings in single quotes.
```

shmsr commented on packages/azure_functions/docs/README.md:
```diff
@@ -0,0 +1,183 @@
+
+`consumer_group` :
+_string_
+ The publish/subscribe mechanism of Event Hubs is enabled through consumer groups. A consumer group is a view (state, position, or offset) of an entire event hub. Consumer groups enable multiple consuming applications to each have a separate view of the event stream, and to read the stream independently at their own pace and with their own offsets.
+Default value: `$Default`
```
```suggestion
 The publish/subscribe mechanism of Event Hubs is enabled through consumer groups. A consumer group is a view (state, position, or offset) of an entire event hub. Consumer groups enable multiple consuming applications to each have a separate view of the event stream, and to read the stream independently at their own pace and with their offsets.
Default value: `$Default`
```

shmsr commented on packages/azure_functions/docs/README.md:
```diff
@@ -0,0 +1,183 @@
+
+`resource_manager_endpoint` :
+_string_
+Optional, by default we are using the azure public environment, to override, users can provide a specific resource manager endpoint in order to use a different azure environment.
```
```suggestion
Optional, by default we are using the Azure public environment, to override, users can provide a specific resource manager endpoint in order to use a different Azure environment.
```

shmsr commented on packages/azure_functions/data_stream/functionapplogs/manifest.yml:
```diff
@@ -0,0 +1,59 @@
+        description: >-
+          Processors are used to reduce the number of fields in the exported event or to enhance the event with metadata. This executes in the agent before the logs are parsed. See [Processors](https://www.elastic.co/guide/en/beats/filebeat/current/filtering-and-enhancing-data.html) for details.
+      - name: sanitize_newlines
+        title: Sanitize New Lines
```
```suggestion
        title: Sanitize new lines
```

Making it sentence case like all other fields

shmsr commented on packages/azure_functions/data_stream/functionapplogs/manifest.yml:
```diff
@@ -0,0 +1,59 @@
+        default: false
+      - name: storage_account_container
+        type: text
+        title: Storage Account Container
```
```suggestion
        title: Storage account container
```

Making it sentence case like all other fields

shmsr reviewed (commented):
Left some very minor comments.

gpop63 commented on packages/azure_functions/data_stream/functionapplogs/elasticsearch/ingest_pipeline/default.yml:
```diff
@@ -0,0 +1,170 @@
+      ignore_missing: true
+  - rename:
+      field: json.properties.functionInvocationId
+      target_field: azure.function.function_invocation_id
```
I wonder if `azure.function.function_invocation_id` is redundant and something like `azure.function.invocation_id` make more sense, this applies to `azure.function.function_name` too.

devamanv replied:
Yes, that's a fair point. I will change that in the next commit.

muthu-mps commented on packages/azure_functions/_dev/build/docs/README.md:
```diff
@@ -0,0 +1,77 @@
+
+
+## Data streams
+Supported log categories:
```
Can we remove the supported log categories here ? This becomes a redundant information when loading the ref link given below. 
Replace the `ref` with `For Reference` or with a complete word.

devamanv replied:
Removed the reference link.

muthu-mps commented on packages/azure_functions/data_stream/functionapplogs/elasticsearch/ingest_pipeline/default.yml:
```diff
@@ -0,0 +1,170 @@
+        - ISO8601
+      if: ctx.json?.time != null
+  - rename:
+      field: json.timeStamp
```
Are we getting the `json.timeStamp `field in the events ? I do see the json.time field which appears to be available in the events as well. If No then can we remove the `rename` and `date` processors ?

devamanv replied:
Fair point, removed the two processors.

muthu-mps commented:
@devamanv - Please find my comments below after installing the enabling the data collection for azure functions.

+ When starting the integration for the first time there is a few minutes delay observed in the data collection. Is this the expected behaviour for a first time installer ?
+ The assets is not available for the integration.The PR description has a mention of having assets along with this PR. Can we create assets.
+ The logo in the Azure function's section of the Azure portal is slightly different than what we have. Can we check and update it.

devamanv commented:
@muthu-mps 

> When starting the integration for the first time there is a few minutes delay observed in the data collection. Is this the expected behaviour for a first time installer ?

Events from Azure are usually slow, and this delay is expected. This may have to do with how the `eventhub` input works under the hood, it needs to start a worker, establish a connection with Azure Eventhub and at the same time store metadata in an instance of the Storage Account. 

> The assets is not available for the integration.The PR description has a mention of having assets along with this PR. Can we create assets

The assets(dashboards) will be created as a separate PR. This task was to add a data stream to start ingesting logs from Azure functions. The assets work is already in progress. I have updated the description accordingly.

> The logo in the Azure function's section of the Azure portal is slightly different than what we have. Can we check and update it.

I have updated the logo to match the one in the Azure Portal, please take a look.

muthu-mps commented on packages/azure_functions/manifest.yml:
```diff
@@ -0,0 +1,81 @@
+format_version: 2.5.1
+name: azure_functions
+title: "Azure Functions"
+version: 0.1.0
```
Can we make the initial release version as `experimental` `0.0.1` until the Kibana dashboard PR is getting merged. Otherwise looks good !

devamanv replied:
Changed the release as experimental and the initial package version to 0.0.1.

zmoog commented:
@cp-elastic uthu-mps

> When starting the integration for the first time there is a few minutes delay observed in the data collection. Is this the expected behaviour for a first time installer ?

To add extra details to what @devamanv said, here's how to check what happens between event creation on Azure and ingestion on Elasticsearch using the document fields.

Here's a KB article I put together a few months ago. AFAIK it's unavailable as a public page, so I'm adding it as a screenshot.

![CleanShot 2023-07-18 at 07 20 19@2x](https://github.com/elastic/integrations/assets/25941/41b00ef7-f8d6-46c9-8888-5f0e93469536)

We can turn it into a public resource if you think it can be helpful to a broader audience.

muthu-mps reviewed (approved):
Looks good !

elasticmachine commented:
Package azure_functions - 0.0.1 containing this change is available at https://epr.elastic.co/search?package=azure_functions
